### ISSUE TITLE
Adapters are counted as standard modules

### ISSUE THREAD

**Issue by mholt** [caddyserver/caddy#5621]
Because [we use a bit of a hack](https://github.com/caddyserver/caddy/blob/7914ba3573fa8eeb05f193fffde1a1c1493e3d0b/caddyconfig/configadapters.go#L121-L137) to make config adapters count as modules, these plugins are incorrectly counted as "standard" (i.e. bundled into the core distribution from this repo).

Since config adapters aren't actually config modules, we should probably not count them as such.

And maybe `list-modules` should be `list-plugins`.

I dunno. Thoughts?

**Comment by mholt** [comment]
@mohammed90 I'm now reconsidering this. I don't think we should count them as normal config modules (because they are not).

Instead, let's reveal which adapters are installed by setting the Accept header in our admin API's `/load` endpoint.

**Comment by mohammed90** [comment]
> [@mohammed90](https://github.com/mohammed90) I'm now reconsidering this. I don't think we should count them as normal config modules (because they are not). Maybe we should rip out the linked code (if possible, if you agree).
> 
> Instead, let's reveal which adapters are installed by setting the Accept header in our admin API's `/load` endpoint.

I'm not sure I follow. Do you mean it'll be in the response `Accept` header when the user calls the endpoint with either GET or HEAD? For reference, we added that code because it wasn't possible to know which adapters are in the build without it.

**Comment by mohammed90** [comment]
> POST with your Content-Type that you want to use

Not sure if you remember, but you implemented support for `Content-Type: text/<adapter-name>`. I like the `HEAD` idea. I still think we should have something in the CLI. Should we add a command `list-adapters`? We probably also should list the adapters in the help text of the `--adapter` flag and the `adapt` command,

**Comment by mholt** [comment]
> Not sure if you remember, but you implemented support for Content-Type: text/<adapter-name>. 

Yeah, but that doesn't tell you what adapters are available. I just am suggesting that if you want to know if your preferred adapter is supported, you can try POSTing with it to begin with, instead of always doing a HEAD first.

What if we put the list of adapters in `build-info`?

**Comment by mohammed90** [comment]
> if your preferred adapter is supported, you can try POSTing with it to begin with, instead of always doing a HEAD first.

That feels wrong. The HEAD idea is very cool. I don't know what I feel about `build-info`. It doesn't seem like the right place.

**Comment by mholt** [comment]
> That feels wrong. The HEAD idea is very cool.

I'm suggesting both should be possible. If your adapter is already installed, then you don't need an extra HEAD request before. Just do the POST. But if the POST fails, you get the Accept header anyway (you don't need HEAD).

HEAD would still be useful if you want to get information only, and you don't (yet) have a config to load.

> I don't know what I feel about build-info. It doesn't seem like the right place.

Hmm. I'm just trying to keep the CLI simple. Why does build-info feel like the wrong place? It should print information about the build. In fact, I think it technically already shows you what packages are compiled in, it just doesn't list _why_ they're compiled in.
